src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -70,6 +68,9 @@ def build(self):
 class StorageProfile(Enum):
     SAPirImage = 1
     SACustomImage = 2
+    ManagedPirImage = 3  # this would be the main scenarios
```

Comment:
There should also be a third SA option: SAExistingImage. #1961.  It can be a later PR though.